### ISSUE TITLE
Graphical representation of face_connections

### ISSUE THREAD

**Issue by jbusecke** [xgcm/xgcm#474]
I am writing the docs for #459, and I am realizing how useful it would be to have a graphical representation like this

<img width="796" alt="image" src="https://user-images.githubusercontent.com/14314623/164062846-619964d0-3a15-4089-b36a-23afa1bf74c4.png">

in the grid.repr. We should be able to create something like this with `graphviz`? as part of a more rich html repr (#276). 

At the very least I think there should be some text representation of which faces are connected?

**Comment by jbusecke** [comment]
Pinging @cisaacstern, who has some experience with making schematics like this?

**Comment by cisaacstern** [comment]
Yes, this is possible with `graphviz`. Here is a very simple graph I embedded in the `pangeo-forge-recipes` docs:

> https://pangeo-forge.readthedocs.io/en/latest/pangeo_forge_cloud/pr_checks_reference.html

Here's where that graph is defined in the docs markdown:

> https://github.com/pangeo-forge/pangeo-forge-recipes/blob/e8468df20aa2b78e0bc56e291b5e7e2bbf07d8e5/docs/pangeo_forge_cloud/pr_checks_reference.md?plain=1#L6-L21

And here's where the sphinx graphviz extension is declared in the docs `conf.py` (which is required to render it):

> https://github.com/pangeo-forge/pangeo-forge-recipes/blob/e8468df20aa2b78e0bc56e291b5e7e2bbf07d8e5/docs/conf.py#L13

This is how I'd develop a more complicated graph:

1. Install this VSCode extension https://marketplace.visualstudio.com/items?itemName=joaompinto.vscode-graphviz 
2. Develop the graph as a `.gv` file in VSCode
3. As you go along, use `CMD SHIFT V` to preview the graph like so
    <img width="800" alt="Screen Shot 2022-04-19 at 10 57 03 AM" src="https://user-images.githubusercontent.com/62192187/164066816-dd9b8e2f-507d-405c-afcb-fb333054c7ff.png">
 
4. When you're happy with it, copy the `.gv` file contents into the docs markdown file as shown in the example link above

**Comment by cisaacstern** [comment]
Hmm I just realized that your aim is to include this in a HTML repr, not sphinx docs. That I haven't done before. And if you're using HTML, I wonder if graphviz is the best choice. It may not be.

**Comment by rabernat** [comment]
Dask's `visualize()` function uses graphviz, so it's certainly not unprecedented.

**Comment by TomNicholas** [comment]
Possibly useful:

https://blog.dask.org/2021/08/23/gsoc-2021-project

**Comment by TomNicholas** [comment]
> Dask's visualize() function uses graphviz, so it's certainly not unprecedented.

Can you put that inside a html repr? If not then it could live under a `Grid.visualize()` method anyway.

Also I like this idea in general!
